### DISCUSSION TITLE
Consume Retail Server APIs in external applications for on-premises

### DISCUSSION THREAD

**Discussion by OmarHeshamMohamed** [microsoft/Dynamics365Commerce.Solutions#192] [Q&A] [answered]
how to consume retail server API's in external applications in on-prem

as the below document is for cloud environment and so far we can't find anything related to on-Primeses deployment 
 
https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/consume-retail-server-api

since the On-primes is communicating through ADFS not Azure and there is no communicating through azure as well, so can you help us understand how we can do the same on on-Primeses.

**Comment by madyke** [comment, marked as answer]
This was addressed internally between support and Sergey, copying Sergey's answer here to close this:

_For all these years we have never been asked for that kind of communication, assuming you are talking about the Application Context in RS, we therefore don’t have any documentation neither samples for that very rare scenario._

_I don’t remember myself trying that at the time we introduced support for ADFS on Channel side, the best you can do is to try it yourself and if you see issues analyze the behavior to see if the PG can help. The starting point for this experiment is to acquire a token from ADFS, rather than from AAD and then send it to RS, analyze warnings/errors in the log file and try to resolve them._
